nvmf-autoconnect/systemd/nvmf-connect-nbft.service.in

```diff
@@ -0,0 +1,11 @@
+# This unit is meant to be started by network management software
+# after a network interface defined in the NBFT gets set up
+[Unit]
+Description=Connect NBFT-defined NVMe-oF subsystems automatically
+ConditionPathExists=|/sys/firmware/acpi/tables/NBFT
+ConditionPathExists=|/sys/firmware/acpi/tables/NBFT1
```

Comment:
I've never seen a `NBFT1` table before.    Is this valid?

```
[root@host-vm ~]# ls /sys/firmware/acpi/tables/NBFT1
ls: cannot access '/sys/firmware/acpi/tables/NBFT1': No such file or directory
[root@host-vm ~]# ls /sys/firmware/acpi/tables/NBFT
/sys/firmware/acpi/tables/NBFT
[root@host-vm ~]# ls /sys/firmware/acpi/tables/
APIC  BGRT  data  DSDT  dynamic  FACP  FACS  HPET  MCFG  NBFT  WAET
[root@host-vm ~]# nvme nbft show
/sys/firmware/acpi/tables/NBFT:

NBFT Subsystems:

Idx|NQN                                                                 |Trsp|Address       |SvcId|HFIs
---+--------------------------------------------------------------------+----+--------------+-----+----
1  |nqn.2014-08.org.nvmexpress:uuid:0c468c4d-a385-47e0-8299-6e95051277db|tcp |192.168.101.20|4420 |1   
2  |nqn.2014-08.org.nvmexpress:uuid:0c468c4d-a385-47e0-8299-6e95051277db|tcp |192.168.110.20|4420 |1   

NBFT HFIs:

Idx|Trsp|PCI Addr  |MAC Addr         |DHCP|IP Addr       |Mask|Gateway |DNS     
---+----+----------+-----------------+----+--------------+----+--------+--------
1  |tcp |0:0:4.0   |ea:eb:d3:58:89:58|no  |192.168.101.30|24  |0.0.0.0 |0.0.0.0 
2  |tcp |0:0:5.0   |ea:eb:d3:59:89:59|no  |192.168.110.30|24  |0.0.0.0 |0.0.0.0 
```

Reply:
> I've never seen a NBFT1 table before. Is this valid?

Should be in cases where a plurality of tables are in XSDT.  That said, there were some instances (maybe with or without the explicit patch[^1]; need to ferret that out) where the order is `NBFT...NBFT2` and some where `NBFT1...NBFT2`.

[^1]: https://github.com/torvalds/linux/commit/6ad90f71129219f9b87ce181189817bcbcde34b8

Reply:
> I've never seen a `NBFT1` table before. Is this valid?

I think so, yes. Per NVMe boot spec, there can be multiple tables with the "NBFT" signature. If this happens, the kernel will expose them in sysfs as `NBFT1`, `NBFT2`, … Notably, there will no `NBFT` entry under `/sys/firmware/acpi/tables` any more, and there won't be `NBFT0`, either. See [the kernel code](https://elixir.bootlin.com/linux/v5.14.21/source/drivers/acpi/sysfs.c#L363). You can verify this behavior on your laptop with `SSDT` tables, for example. Most systems have multiple `SSDT` tables, and the kernel presents them as `SSDT1`, `SSDT2`, etc. 

The way I read the code, either `NBFT` or `NBFT1` are always present; the former if there's just one NBFT, and the latter if there are many.

You are welcome to discuss this with ACPI experts at Red Hat. I discussed it with one of our ACPI experts at SUSE already, and he agreed with it.